src/modules/user/user.usecase.ts

```diff
@@ -518,4 +555,36 @@ export class UserUseCases {
       password,
     });
   }
+
+  async updatePassword(
+    user: User,
+    updatePasswordDto: UpdatePasswordDto,
+  ): Promise<void> {
+    const { currentPassword, newPassword, newSalt, mnemonic, privateKey } =
+      updatePasswordDto;
+
+    if (user.password.toString() !== currentPassword) {
+      throw new UnauthorizedException();
+    }
+    await this.userRepository.updateById(user.id, {
+      password: newPassword,
+      hKey: Buffer.from(newSalt),
```

Comment:
Why? Check [this](https://github.com/internxt/drive-server/blob/dca8b2b42d1510cd30f71efd6e67e0c504307856/src/app/services/user.js#L342)